docs/external-node.md

```diff
@@ -0,0 +1,305 @@
+  resourceVersion: "5513"
+  uid: 5f360f32-7806-4d2d-9f36-80ce7db8de10
+spec:
+  endpoints:
+    - ip: 172.16.100.3
+  externalNode: vm1
+```
+
+Note: **Only one interface is supported for Antrea v1.8**.
+
+## Apply Antrea NetworkPolicy to ExternalNode
+
+Antrea NetworkPolicy applies to an `ExternalNode` by setting the `appliedTo` field with a `externalEntitySelector`.
+The `ExternalNode` object is converted to
+`ExternalEntities`, which are used by `antrea-controller` to calculate the NetworkPolicy span, and `antrea-agent`
+running on the `ExternalNode` receives the internal Antrea NetworkPolicy objects.
+
+Following types of remote addresses are supported when applying Antrea NetworkPolicy to an external Node:
+
+- ExternalNode with a `externalEntitySelector`
+- CIDR with an `ipBlock`
+- fqdn address in an egress rule
+
+Following actions are supported when applying Antrea NetworkPolicy to an external Node:
+
+- Allow
+- Drop
+- Reject
+
+Following is an example to apply Antrea NetworkPolicy to the `ExternalNode` objects labeled with `role=db` to reject the
+SSH connections from IP "172.16.100.5" or from other external Nodes labeled with `role=front`:
+
+```yaml
+kind: NetworkPolicy
+metadata:
+  name: anp1
+  namespace: vm-ns
+spec:
+  priority: 9000.0
+  appliedTo:
+    - externalEntitySelector:
+        matchLabels:
+          role: db
+  egress:
+    - action: Reject
+      ports:
+        - protocol: TCP
+          port: 22
+      to:
+        - externalEntitySelector:
+            matchLabels:
+              role: front
+        - ipBlock:
+            cidr: 172.16.100.5/32
+```
+
+`antrea-agent` running on the target external Node sets up a connection with
+`antrea-controller` using the `ExternalNode` name as an option to receive the internal Antrea NetworkPolicy objects.
+More details please refer to the
+[installation guide](vm-installation.md) for `antrea-agent` on an external Node.
+
+`antrea-agent` realizes the Antrea NetworkPolicy on OVS using OpenFlow entries.
+
+## Security
+
+An external Node is thought as an untrusted point on the network. To follow the security principle that least
+permissions are given, the RBAC for `antrea-agent`
+running on `ExternalNode` feature is modified:
+
+- Only `get`, `list` and `watch` permissions are given on resource `ExternalNode`
+- Only `update` permission is given on resource `antreaagentinfos`, and `create`
+  permission is moved to `antrea-controller`
+- for more details please refer to [vm-agent.yaml](../build/yamls/externalnode/vm-agent.yaml)
+
+`antrea-agent` reports its status by updating the `antreaagentinfo` resource which is configured with the same name as
+the `ExternalNode`. `antrea-controller`
+creates an `antreaagentinfo` object for each new `ExternalNode`, and then `antrea-agent`
+updates it every minute with its latest status. `antreaagentinfo` is deleted by
+`antrea-controller` when the `ExternalNode` is removed.
+
+## Traffic model
+
+### OVS pair ports
```

Comment:
Probably name it "VM network interface configuration" or "VM network configuration" and make it the top level.

I am assuming all contents can be interested by regular users. If you feel there are design details users do not care we should separate them out.

Reply:
Renamed as "VM network configuration", and move as  a separate section.